src/UnderlineNav2/UnderlineNav.tsx

```diff
@@ -0,0 +1,161 @@
+import React, {useRef, forwardRef, useCallback, useState, MutableRefObject, RefObject} from 'react'
+import Box from '../Box'
+import {merge, SxProp, BetterSystemStyleObject} from '../sx'
+import {UnderlineNavContext} from './UnderlineNavContext'
+import {ActionMenu} from '../ActionMenu'
+import {ActionList} from '../ActionList'
+import {useResizeObserver, ResizeObserverEntry} from '../hooks/useResizeObserver'
+
+type Overflow = 'auto' | 'menu' | 'scroll'
+type ChildWidthArray = Array<{width: number}>
+type ResponsiveProps = {
+  items: Array<React.ReactElement>
+  actions: Array<React.ReactElement>
+}
+const overflowEffect = (
+  width: number,
+  childArray: Array<React.ReactElement>,
+  childWidthArray: ChildWidthArray,
+  callback: (props: ResponsiveProps) => void
+) => {
+  if (childWidthArray.length === 0) {
+    callback({items: childArray, actions: []})
+  }
+
+  // do this only for overflow
+  const numberOfItemsPossible = calculatePossibleItems(childWidthArray, width)
+  const items: Array<React.ReactElement> = []
+  const actions: Array<React.ReactElement> = []
+
+  // eslint-disable-next-line github/array-foreach
+  childArray.forEach((child: React.ReactElement, index: number) => {
```

Comment:
Why do we have to use `forEach` instead of a [for...of loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of)?

Reply:
I just found it convenient to get the `index` in this case. But I guess you are right. This is an eslint disable I can remove from code.

Reply:
done!